### PR TITLE
[Bugfix] Enhancement compatibility for create Materialized View

### LINKED ISSUE #10318: FE crash when create materialized view
### Steps to reproduce the behavior (Required)
UNKNWON
### Expected behavior (Required)
Create OK
### Real behavior (Required)
2022-08-22 12:12:59,944 WARN (UNKNOWN 172.31.26.88_9010_1659950384187(-1)|1) [GlobalStateMgr.notifyNewFETypeTransfer():1590] notify new FE type transfer: UNKNOWN
2022-08-22 12:13:01,957 WARN (RepNode 172.31.26.88_9010_1659950384187(-1)|52) [GlobalStateMgr.notifyNewFETypeTransfer():1590] notify new FE type transfer: MASTER
2022-08-22 12:13:04,054 ERROR (stateListener|66) [BDBJournalCursor.next():117] fail to read journal entity key=808178, will exit
com.google.gson.JsonIOException: java.io.IOException: error happens when parsing create materialized view stmt: CREATE MATERIALIZED VIEW xxxx
        at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:287) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.persist.gson.RuntimeTypeAdapterFactory$1.read(RuntimeTypeAdapterFactory.java:367) ~[starrocks-fe.jar:?]
        at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:932) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:897) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:846) ~[spark-dpp-1.0.0.jar:?]
        at com.google.gson.Gson.fromJson(Gson.java:817) ~[spark-dpp-1.0.0.jar:?]
        at com.starrocks.alter.AlterJobV2.read(AlterJobV2.java:251) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.JournalEntity.readFields(JournalEntity.java:543) ~[starrocks-fe.jar:?]
        at com.starrocks.journal.bdbje.BDBJournalCursor.next(BDBJournalCursor.java:115) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.replayJournal(GlobalStateMgr.java:1721) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.transferToMaster(GlobalStateMgr.java:916) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr.access$900(GlobalStateMgr.java:259) [starrocks-fe.jar:?]
        at com.starrocks.server.GlobalStateMgr$5.runOneCycle(GlobalStateMgr.java:1648) [starrocks-fe.jar:?]
        at com.starrocks.common.util.Daemon.run(Daemon.java:115) [starrocks-fe.jar:?]
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.3

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #10318

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedIndexMeta.java [hunk 1]
```diff
@@ -187,15 +195,33 @@ public void gsonPostProcess() throws IOException {
...
             setColumnsDefineExpr(columnNameToDefineExpr);
         } catch (Exception e) {
-            throw new IOException("error happens when parsing create materialized view stmt: " + defineStmt.originStmt, e);
+            fallback = true;
+            LOG.warn("error happens when parsing create materialized view stmt in meta: " +
+                    defineStmt.originStmt, e);
         }
-    }
 
+        if (fallback) {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedIndexMeta.java:211]: how about adding this into MVUtils

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 9 / 18 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/mv/MVUtils.java|5|14|35.71%|[94, 95, 99, 101, 103, 104, 105, 106, 107]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndexMeta.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/RollupJobV2.java|2|2|100.00%|[]|

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Review by caneGuy — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MVUtils.java [hunk 2]
```diff
@@ -64,4 +82,31 @@ public static String getMVColumnName(Column mvColumn, String functionName, Strin
         }
         return "mv_" + mvColumn.getAggregationType().name().toLowerCase() + "_" + queryColumn;
     }
+
+    public static Map<String, Expr> parseColumnNameToDefineExpr(OriginStatement originStmt) throws IOException {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/mv/MVUtils.java:86]: why put this util function in optimizer rule? it seems that this api is used by catalog?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 9 / 18 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/MetaUtils.java|5|14|35.71%|[111, 112, 116, 118, 120, 121, 122, 123, 124]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndexMeta.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/RollupJobV2.java|2|2|100.00%|[]|

**Review by meegoo — APPROVED** [review]

**Review by caneGuy — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.4 branch-2.3

**Pull request merged**